### DISCUSSION TITLE
One renovate per package per dependency

### DISCUSSION THREAD

**Discussion by zaffnet** [renovatebot/renovate#19198] [Request Help]
Hi everyone, I am new to `renovate.` I am trying to add `renovate` to a repository. The repository structure is as follows:

```
repo
    \_ internal-package-0 (depends on: dep0, dep1, and internal-package-2)
    \_ internal-package-1 (depends on: dep1, dep2, and internal-package-0)
    \_ internal-package-2 (depends on: dep0, dep1, dep2)
    ...
```

These internal packages are in python and are managed by poetry (have a pyproject.toml and poetry.lock files). I also have tests in CircleCI for each package.

 I want to create `renovate` PRs to update the dependencies of these packages. My constraints are:
 1. create one PR per dependency per internal package.
 2. never create a PR if the CI test corresponding to the internal package fails.
 3. auto-rebase PRs with the `main` branch. If rebase fails, close the PR.
 4. create at most 10 PRs per day. Prioritize it based on vulnerability/risk.

How should I write my `renovate` config for my use case?

Thanks a lot!

**Comment by rarkins** [comment]
1. Set `additionalBranchPrefix` to be `"{{parentDir}}"` so that PRs are split per internal package directory name
2. I'm not sure quite what you mean, but perhaps prCreation=success is what you want (it means PRs won't be created unless all tests pass)
3. Sounds like you want rebaseWhen=behind-base-branch. Rebasing is unlikely to fail, but if it results in zero changes then the PR will be closed
4. There is no per day limit, but there is concurrent and hourly limits you can find in the docs
